File: Cpp/Longest Common Prefix.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string longestCommonPrefix(vector<string> &strs) {
    int len=strs.size();
    string str="";
    if(len==0) return str;
    sort(strs.begin(),strs.end());
    string str1=strs[0],str2=strs[len-1];
    for(int i=0;i<str1.size()&&i<str2.size();i++){
        if(str1[i]==str2[i]) str=str+str1[i];
        else break;
    }
    return str;
}

File: Python/Longest-Common-Prefix.py
#!/usr/bin/env python


def longestCommonPrefix(self, strs):
    s = ''
    num = 0
    if len(strs) == 0:
        return s
    elif len(strs) == 1:
        return strs[0]
    while num >= 0:
        if num >= len(strs[0]):
            return s
        ch = strs[0][num]
        for i in range(1, len(strs)):
            if num >= len(strs[i]):
                return s
            elif strs[i][num] != ch:
                return s
        s += ch
        num += 1
    return s